src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/InstructionDecoderTests.cs

```diff
@@ -402,6 +402,144 @@ class C
             Assert.Equal("System.Action<System.Func<object>>", GetReturnTypeName(source, "C.M1", [typeof(object)]));
         }
 
+        [Fact]
+        public void GetCompactName_Members()
+        {
+            var source = """
+                using System;
+                namespace System.Runtime.CompilerServices
+                {
+                    public class IsExternalInit { }
+                }
+                class C
+                {
+                    static C() { }
+                    C(int x) { }
+                    ~C() { }
+                    object F() => null;
+                    object P1 { get; }
+                    object P2 { set { } }
+                    object P3 { init { } }
+                    object this[int i] { get { return null; } set { } }
+                    event EventHandler E;
+                    public static C operator+(C c) => c;
+                    public static implicit operator int(C c) => 0;
+                    public static explicit operator string(C c) => "";
+                }
+                static class E
+                {
+                    static void M(this string x) { }
+                }
+                """;
+            var compilation = CreateCompilation(source);
+            var containingType = compilation.GlobalNamespace.GetTypeMember("C");
+            VerifyMethodName(compilation, containingType.GetMethod("F"), "C.F()", "F");
+            VerifyMethodName(compilation, containingType.GetMethod("get_P1"), "C.P1.get()", "P1");
+            VerifyMethodName(compilation, containingType.GetMethod("set_P2"), "C.P2.set(value)", "P2");
+            VerifyMethodName(compilation, containingType.GetMethod("set_P3"), "C.P3.init(value)", "P3");
+            VerifyMethodName(compilation, containingType.GetMethod("get_Item"), "C.this[int].get(i)", "this");
```

Comment:
>this [](http://example.com/codeflow?start=103&length=4)

nit: I'm a bit concerned about the indexer case. `this` doesn't really shout "indexer"...

Reply:
Replaced with `this[]`.